src/Serilog/ILogger.cs

```diff
@@ -179,15 +323,43 @@ public interface ILogger
         /// <param name="messageTemplate">Message template describing the event.</param>
         /// <param name="propertyValues">Objects positionally formatted into the message template.</param>
         [MessageTemplateFormatMethod("messageTemplate")]
-        void Write(LogEventLevel level, Exception exception, string messageTemplate, params object[] propertyValues);
+#if FEATURE_DEFAULT_INTERFACE
+        [CustomDefaultMethodImplementation]
+#endif
+        void Write(LogEventLevel level, Exception exception, string messageTemplate, params object[] propertyValues)
+#if FEATURE_DEFAULT_INTERFACE
+        {
+            if (!IsEnabled(level)) return;
+            if (messageTemplate == null) return;
+
+            // Catch a common pitfall when a single non-object array is cast to object[]
+            if (propertyValues != null &&
+                propertyValues.GetType() != typeof(object[]))
+                propertyValues = new object[] { propertyValues };
+
+            if (DefaultLoggerImpl.BindMessageTemplate(messageTemplate, propertyValues, out var parsedTemplate, out var boundProperties))
```

Comment:
Should this be `this.BindMessageTemplate()`?

Reply:
right, fixed